src/frontend/src/lib/services/address.services.ts

```diff
@@ -147,26 +160,43 @@ const loadIdbTokenAddress = async <T extends Address>({
 			`Error encountered while searching for locally stored ${tokenId.description} public address in the browser.`
 		);
 
-		return { success: false };
+		return { success: false, err: new AddressError(tokenId) };
 	}
 
 	return { success: true };
 };
 
-export const loadIdbBtcAddressMainnet = async (): Promise<ResultSuccess> =>
+const loadIdbBtcAddressMainnet = async (): Promise<ResultSuccess<AddressError>> =>
 	loadIdbTokenAddress<BtcAddress>({
 		tokenId: BTC_MAINNET_TOKEN_ID,
 		getIdbAddress: getIdbBtcAddressMainnet,
 		updateIdbAddressLastUsage: updateIdbBtcAddressMainnetLastUsage
 	});
 
-export const loadIdbEthAddress = async (): Promise<ResultSuccess> =>
+const loadIdbEthAddress = async (): Promise<ResultSuccess<AddressError>> =>
 	loadIdbTokenAddress<EthAddress>({
 		tokenId: ETHEREUM_TOKEN_ID,
 		getIdbAddress: getIdbEthAddress,
 		updateIdbAddressLastUsage: updateIdbEthAddressLastUsage
 	});
 
+export const loadIdbAddresses = async (): Promise<ResultSuccess<TokenId[]>> => {
```

Comment:
Shouldn't we bubble `ResultSuccess<AddressError[]>` to the component? That way in the component `error` has also a strict type?

Reply:
for sure!